docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst

```diff
@@ -0,0 +1,616 @@
++---------------------------------------------------------------------------------------------------------------------------------+---------------------+--------------------------------+---------+
+| ``case when`` Expr_ ``then`` StmtSequence_ «``else`` StmtSequence_»? ``end``                                                    | CaseExpr_           | ControlExpr_                   |         |
++---------------------------------------------------------------------------------------------------------------------------------+---------------------+--------------------------------+---------+
+| ``case`` Expr_ ``in`` «TERM CaseExpr_»+ ``end`` f                                                                               | CaseExpr_           | ControlExpr_                   |         |
++---------------------------------------------------------------------------------------------------------------------------------+---------------------+--------------------------------+---------+
+
+Unary operations
+~~~~~~~~~~~~~~~~
+
+All classes in this subsection are subclasses of UnaryOperation_.
+
++--------------------+----------------+----------------------------+-------------------+
+| Expression syntax  |  CodeQL class  | Superclasses               | Remarks           |
++====================+================+============================+===================+
+| ``~`` Expr_        | ComplementExpr_ | UnaryBitwiseOperation_    |                   |
++--------------------+----------------+----------------------------+-------------------+
+| ``defined?`` Expr_ | DefinedExpr_    | UnaryOperation_           |                   |
++--------------------+----------------+----------------------------+-------------------+
+| ``**`` Expr_       | HashSplatExpr_  | UnaryOperation_           |                   |
++--------------------+----------------+----------------------------+-------------------+
+| ``!`` Expr_        | NotExpr_        | UnaryOperation_           |                   |
++--------------------+----------------+----------------------------+-------------------+
+| ``not`` Expr_      | NotExpr_        | UnaryOperation_           |                   |
++--------------------+----------------+----------------------------+-------------------+
+| ``*`` Expr_        | SplatExpr_      | UnaryOperation_           |                   |
++--------------------+----------------+----------------------------+-------------------+
+| ``-`` Expr_        | UnaryMinusExpr_ | UnaryArithmeticOperation_ |                   |
++--------------------+----------------+----------------------------+-------------------+
+| ``+`` Expr_        | UnaryPlusExpr_  | UnaryArithmeticOperation_ |                   |
++--------------------+----------------+----------------------------+-------------------+
```

Comment:
This table formatting looks as if it's off by one, resulting in a display with columns 2 and 3 merged. I _think_ this will fix it:

```suggestion
+--------------------+-----------------+---------------------------+-------------------+
| Expression syntax  |  CodeQL class  | Superclasses               | Remarks           |
+====================+=================+===========================+===================+
| ``~`` Expr_        | ComplementExpr_ | UnaryBitwiseOperation_    |                   |
+--------------------+-----------------+---------------------------+-------------------+
| ``defined?`` Expr_ | DefinedExpr_    | UnaryOperation_           |                   |
+--------------------+-----------------+---------------------------+-------------------+
| ``**`` Expr_       | HashSplatExpr_  | UnaryOperation_           |                   |
+--------------------+-----------------+---------------------------+-------------------+
| ``!`` Expr_        | NotExpr_        | UnaryOperation_           |                   |
+--------------------+-----------------+---------------------------+-------------------+
| ``not`` Expr_      | NotExpr_        | UnaryOperation_           |                   |
+--------------------+-----------------+---------------------------+-------------------+
| ``*`` Expr_        | SplatExpr_      | UnaryOperation_           |                   |
+--------------------+-----------------+---------------------------+-------------------+
| ``-`` Expr_        | UnaryMinusExpr_ | UnaryArithmeticOperation_ |                   |
+--------------------+-----------------+---------------------------+-------------------+
| ``+`` Expr_        | UnaryPlusExpr_  | UnaryArithmeticOperation_ |                   |
+--------------------+-----------------+---------------------------+-------------------+
```

Reply:
Thanks for catching this, I kept having trouble with table formatting in this PR.